accesstoken2/src/services/jwt-verifier.ts

```diff
@@ -0,0 +1,55 @@
+import { importJWK, JWTPayload, jwtVerify } from "jose";
+import { ConfigService } from "./config-service";
+import { JWTVerifyOptions } from "jose/dist/types/jwt/verify";
+export class JwtVerifier {
+    constructor(private configService: ConfigService) {}
+
+    public async verify(
+        encodedJwt: any,
+        clientId: string,
+        mandatoryClaims: Set<string>,
+        expectedClaimValues: Map<string, string>,
+    ): Promise<JWTPayload> {
+        const signingPublicJwkBase64 = await this.configService.getPublicSigningJwk(clientId);
+        const signingPublicJwk = JSON.parse(Buffer.from(signingPublicJwkBase64, "base64").toString("utf8"));
+        const signingAlgorithm = await this.configService.getJwtSigningAlgorithm(clientId);
```

Comment:
Some minor performance improvements. 

we have two Promises which are using the same value to return something from the config service.

A promise is a blocking action, but in the world of NodeJS we can execute these 'simultaneously'...  (its not actually simultaneous, but it will put them on the queue one after the over so in theory it should be faster).

To do so we do this.

```
const signingPublicJwkBase64Promise = this.configService.getPublicSigningJwk(clientId);
const signingAlgorithmPromise = this.configService.getJwtSigningAlgorithm(clientId)
const [signingPublicJwkBase64, signingAlgorithm] = Promise.all([signingPublicJwkBase64Promise, signingAlgorithmPromise]);
```